modules/imgproc/src/sumpixels.simd.hpp

```diff
@@ -180,61 +420,690 @@ struct Integral_SIMD<uchar, float, double>
+
+                v_float32 prev = vx_setzero_f32();
+                int j = 0;
+                for ( ; j + v_uint16::nlanes <= width; j += v_uint16::nlanes)
+                {
+                    v_int16 el8 = v_reinterpret_as_s16(vx_load_expand(src_row + j));
+                    v_float32 el4l, el4h;
+#if CV_AVX2 && CV_SIMD_WIDTH == 32
+                    __m256i vsum = _mm256_add_epi16(el8.val, _mm256_slli_si256(el8.val, 8));
+                    el4l.val = _mm256_add_ps(_mm256_cvtepi32_ps(_mm256_cvtepi16_epi32(_v256_extract_low(vsum))), prev.val);
+                    el4h.val = _mm256_add_ps(_mm256_cvtepi32_ps(_mm256_cvtepi16_epi32(_v256_extract_high(vsum))), _mm256_permute2f128_ps(el4l.val, el4l.val, 0x31));
+                    prev.val = _mm256_permute2f128_ps(el4h.val, el4h.val, 0x31);
+#else
+#if CV_SIMD_WIDTH >= 32
+                    el8 += v_rotate_left<4>(el8);
+#if CV_SIMD_WIDTH == 64
+                    el8 += v_rotate_left<8>(el8);
+#endif
+#endif
+                    v_int32 el4li, el4hi;
+                    v_expand(el8, el4li, el4hi);
+                    el4l = v_cvt_f32(el4li) + prev;
+                    el4h = v_cvt_f32(el4hi) + el4l;
+#if CV_SIMD_WIDTH == 16
+                    prev = el4h;
+#elif CV_SIMD_WIDTH == 32
+                    prev = v_combine_high(el4h, el4h);
+#else
+                    v_float32 t = v_rotate_right<12>(el4h);
+                    t |= v_rotate_left<4>(t);
+                    prev = v_combine_low(t, t);
+#endif
+#endif
+                    v_store(sum_row + j                    , el4l + vx_load(prev_sum_row + j                    ));
+                    v_store(sum_row + j + v_float32::nlanes, el4h + vx_load(prev_sum_row + j + v_float32::nlanes));
+                }
+
+                for (float v4 = sum_row[j - 1] - prev_sum_row[j - 1],
+                           v3 = sum_row[j - 2] - prev_sum_row[j - 2],
+                           v2 = sum_row[j - 3] - prev_sum_row[j - 3],
+                           v1 = sum_row[j - 4] - prev_sum_row[j - 4]; j < width; j += 4)
+                {
+                    sum_row[j]     = (v1 += src_row[j])     + prev_sum_row[j];
+                    sum_row[j + 1] = (v2 += src_row[j + 1]) + prev_sum_row[j + 1];
+                    sum_row[j + 2] = (v3 += src_row[j + 2]) + prev_sum_row[j + 2];
+                    sum_row[j + 3] = (v4 += src_row[j + 3]) + prev_sum_row[j + 3];
+                }
+            }
+        }
+        else
+        {
+            return false;
+        }
+        vx_cleanup();
+
+        return true;
+    }
+};
+
+#if CV_SIMD128_64F && !CV_AVX512_SKX
+template <>
+struct Integral_SIMD<uchar, double, double>
+{
+    Integral_SIMD() {}
+
+    bool operator()(const uchar * src, size_t _srcstep,
+        double * sum, size_t _sumstep,
+        double * sqsum, size_t,
+        double * tilted, size_t,
+        int width, int height, int cn) const
+    {
```

Comment:
I think call to specific AVX512 implementation could be moved to the begging of this implementation with proper check for requested mode
```
#if CV_AVX512_SKX
if (!tilted && cn <= 4 && (cn > 1 || sqsum))
{
    calculate_integral_avx512(src, _srcstep, sum, _sumstep, sqsum, _sqsumstep, width, height, cn);
    return true;
}
#endif
```